workalendar/tests/test_registry_africa.py

```diff
@@ -23,3 +25,14 @@ def test_africa(self):
         self.assertIn(Madagascar, classes)
         self.assertIn(SaoTomeAndPrincipe, classes)
         self.assertIn(SouthAfrica, classes)
+
+
+class SouthAfricaTest(GenericCalendarTest):
+    cal_class = SouthAfrica
+
+    def test_south_africa_2019(self):
+        holidays = self.cal.holidays_set(2019)
+
+        # variable days
+        self.assertIn(date(2019, 5, 8), holidays)  # 2019 National Elections
+        self.assertNotIn(date(2017, 5, 8), holidays)
```

Comment:
hem... no it won't work like this. You're fetching the "2019" holidays, and of course, the 2017-05-08 is not in the holiday set.

Let me suggest something:

```suggestion
        holidays_2017 = self.cal.holidays_set(2017)
        self.assertNotIn(date(2017, 5, 8), holidays_2017)
```